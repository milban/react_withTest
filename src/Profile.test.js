import React from "react"
//import { mount } from "enzyme"
import { render } from "@testing-library/react"
import Profile from "./Profile"

describe("<Profile />", () => {
  it("matches snapshot", () => {
    const utils = render(<Profile username="milban" name="박준혁" />)
    expect(utils.container).toMatchSnapshot()
  })
  it("shows the props correctly", () => {
    const utils = render(<Profile username="milban" name="박준혁" />)
    utils.getByText("milban")
    utils.getByText("(박준혁)")
    utils.getByText(/박/)
  })
})

// describe("<Profile />", () => {
//   it("matches spanpshot", () => {
//     const wrapper = mount(<Profile username="milban" name="박준혁" />)
//     expect(wrapper).toMatchSnapshot()
//   })
//   it("renders username and name", () => {
//     const wrapper = mount(<Profile username="milban" name="박준혁" />)

//     expect(wrapper.props().username).toBe("milban")
//     expect(wrapper.props().name).toBe("박준혁")

//     const boldElement = wrapper.find("b")
//     expect(boldElement.contains("milban")).toBe(true)
//     const spanElement = wrapper.find("span")
//     expect(spanElement.text()).toBe("(박준혁)")
//   })
// })
