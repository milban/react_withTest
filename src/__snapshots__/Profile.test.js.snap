// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`<Profile /> matches snapshot 1`] = `
<div>
  <div>
    <b>
      milban
    </b>
    <span>
      (
      박준혁
      )
    </span>
  </div>
</div>
`;
